lib/backends/sdc/containers.js

```diff
@@ -3578,6 +3578,14 @@ function _runExec(opts, callback) {
 
     serverSocket.on('connect', setupListeners);
 
+    // error can happen before connect too (eg. ECONNREFUSED)
+    serverSocket.on('error', function (error) {
+        opts.log.error('serverSocket for %s threw an error %s',
+            cmdString, error.toString());
+
+        endSocket(error);
```

Comment:
> Does changing this to not call endSocket and instead just doing:

    // error can happen before connect too (eg. ECONNREFUSED)
    serverSocket.on('error', function _onServerSocketError(error) {
        opts.log.error('serverSocket for %s threw an error %s',
            cmdString, error.toString());

        serverSocket.end();
        cb(error);
    });
make more sense?

Yes, and I don't think the call to `serverSocket.end()` is needed here too.